api/src/main/java/org/apache/iceberg/view/ViewRepresentation.java

```diff
@@ -18,21 +18,16 @@
  */
 package org.apache.iceberg.view;
 
-import java.util.Locale;
+import org.immutables.value.Value;
 
+@Value.Immutable
 public interface ViewRepresentation {
 
-  enum Type {
-    SQL;
+  class Type {
+    private Type() {}
 
-    public static Type fromString(String typeName) {
-      return valueOf(typeName.toUpperCase(Locale.ENGLISH));
-    }
-
-    public String typeName() {
-      return name().toLowerCase(Locale.ENGLISH);
-    }
+    public static final String SQL = "sql";
```

Comment:
Why change this from an enum to a String?

Reply:
I made the suggestion since I see some other enum like classes are also implemented directly as strings, such as `DataOperations`, and it seems to simplify the code a bit. But if there is a specific reason for using enum we can stick to that.

Reply:
I think it depends on whether we want to use this enum in switch statements in our code or if we want to extend it. For example, we could have a reference to the parser in the enum so we look up the symbol and then call something like `ViewRepresentation.SQL.parse(jsonNode)`.

Since we only have the parser selection right now, it doesn't seem like it matters much.